### ISSUE TITLE
Step 6 - Code snippet does not work

### ISSUE THREAD

**Issue by bahamafive** [coopengo/tryton-training#11]
This is a marvelous tutorial! I stumbled over many missing links (all links to documentation are broken) and some typos... but nevertheless: This tutorial is great!

In Step6 though I found a snippet that does not work at all - and I don't know how to fix (tryton 5.4):

https://github.com/coopengo/tryton-training/wiki/5.0-step6
In the section Validate:

> @classmethod
> def __setup__(cls):
>     super().__setup__()
>     cls._error_messages.update({
>             'invalid_isbn': 'ISBN should only be digits',
>             })
> 
> @classmethod
> def validate(cls, books):
>     for book in books:
>         if not book.isbn:
>             continue
>         try:
>             if int(book.isbn) < 0:
>                 raise ValueError
>         except ValueError:
>             cls.raise_user_error('invalid_isbn')

Thanks for your tutorial - I am enjoying it!
Kind regards,
Marco

**Comment by bahamafive** [comment]
Found the reason here: https://discuss.tryton.org/t/tryton-release-5-2/1294

Until now, all the errors were raised using the two same exceptions UserError and UserWarning. 
This was not great for testing nor to catch a specific error.
So we removed the methods raise_user_error and raise_user_warning to replace them by the Python raise statement 5. This allows to use custom exceptions inheriting from one of the two bases.
As we need to set on the exception the error message for the user, we created a new model to store all the messages ir.message. It can be used with the new gettext 1 tool to get the message in the language of the user.

**Comment by JCavallo** [comment]
Yeah, the latest version for now is 5.0, and it is the only major modification that affected it in 5.2

I'll probably add a note on this, until I actually create a new version
